fenix/app/src/androidTest/java/org/mozilla/fenix/experimentintegration/SurveyExperimentIntegrationTest.kt

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.experimentintegration
+
+import android.widget.TextView
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiSelector
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.helpers.AndroidAssetDispatcher
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestHelper
+import org.mozilla.fenix.ui.robots.*
+
+
+class SurveyExperimentIntegrationTest {
+    private lateinit var mDevice: UiDevice
+    private lateinit var mockWebServer: MockWebServer
+
+    @get:Rule
+    val activityTestRule = HomeActivityTestRule()
+
+    @Before
+    fun setUp() {
+        mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        mockWebServer = MockWebServer().apply {
+            dispatcher = AndroidAssetDispatcher()
+            start()
+        }
+        TestHelper.appContext.settings().showSecretDebugMenuThisSession = true
+    }
+
+    @After
+    fun tearDown() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openExperimentsMenu{
+            getExperiment("Viewpoint")
+        }
```

Comment:
It's generally not recommended to put UI actions in the tearDown as it's mainly used for resource cleanup. The main reason for this is that `tearDown` is called after every test method, even if the test fails. If your test fails in a way that leaves the UI in an unexpected state, any UI actions in the `tearDown` method may also fail, causing further confusion when trying to diagnose what went wrong with the test. Furthermore, if a UI action in the `tearDown` method fails for any reason, it might prevent other cleanup actions from being executed,